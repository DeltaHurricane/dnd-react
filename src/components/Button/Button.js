import React from 'react';
import PropTypes from 'prop-types';
import './Button.scss';

export default function Button({ name }) {
  return (
    <button
      className="button button--flash button--confirmation"
      type="button"
    >
      {name}
    </button>
  );
}

Button.propTypes = {
  name: PropTypes.string.isRequired,
};
