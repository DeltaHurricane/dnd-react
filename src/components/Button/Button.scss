.button {
  width: 80px;
  height: 35px;
  padding: 1%;
  border-radius: 20% 20% 20% 20%;
  margin-top: 20px;
}

.button--confirmation {
  position: relative;
}

.button--flash {
  overflow: hidden;
  transition: 1s;
}

.button:hover {
  background-color: rgb(185, 63, 63);
  box-shadow: -1em 0.35em 0.9em 0.1em rgba(0, 0, 0, 0.63);
}

.button--flash::after {
  content: '';
  position: absolute;
  top: 0;
  left: -35%;
  width: 20%;
  height: 100%;
  background: rgba(255, 255, 255, 0.3);
  transform: skewX(-30deg);
  transition: 0.5s;
}

.button--flash:hover::after {
  left: 125%;
}
