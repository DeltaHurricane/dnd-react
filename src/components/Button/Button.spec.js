import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import Button from './Button';

const wrapper = (props = {}) => shallow(<Button {...props} />);

describe('should click', () => {
  it('should render without throwing an error', () => {
    const props = {
      name: 'test',
      onClick: () => 'teste',
    };
    expect(wrapper(props)).toMatchSnapshot();
  });
  it('should contain button', () => {
    const callback = sinon.spy();
    const props = {
      name: 'test',
      onClick: callback,
    };
    wrapper(props).find('button').simulate('click');
    expect(callback.calledOnce).toEqual(true);
  });
});
