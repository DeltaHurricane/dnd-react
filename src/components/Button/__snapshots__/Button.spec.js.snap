// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should click should render without throwing an error 1`] = `
<button
  className="button button--flash button--confirmation"
  onClick={[Function]}
  type="button"
>
  test
</button>
`;
